official/transformer/compute_bleu.py

```diff
@@ -109,21 +111,22 @@ def main(unused_argv):
 
 
 if __name__ == "__main__":
-  parser = argparse.ArgumentParser()
-  parser.add_argument(
-      "--translation", "-t", type=str, default=None, required=True,
-      help="[default: %(default)s] File containing translated text.",
-      metavar="<T>")
-  parser.add_argument(
-      "--reference", "-r", type=str, default=None, required=True,
-      help="[default: %(default)s] File containing reference translation",
-      metavar="<R>")
-  parser.add_argument(
-      "--bleu_variant", "-bv", type=str, choices=["uncased", "cased"],
-      nargs="*", default=None,
-      help="Specify one or more BLEU variants to calculate (both are "
-           "calculated by default. Variants: \"cased\" or \"uncased\".",
-      metavar="<BV>")
-
-  FLAGS, unparsed = parser.parse_known_args()
-  main(sys.argv)
+  flags.DEFINE_string(
+      name="translation", default=None,
+      help=flags_core.help_wrap("File containing translated text."))
+  flags.mark_flag_as_required("translation")
+
+  flags.DEFINE_string(
+      name="reference", default=None,
+      help=flags_core.help_wrap("File containing reference translation."))
+  flags.mark_flag_as_required("reference")
+
+  flags.DEFINE_multi_enum(
+      name="bleu_variant", short_name="bv", default=["uncased", "cased"],
+      enum_values=["uncased", "cased"], case_sensitive=False,
+      help=flags_core.help_wrap(
+          "Specify one or more BLEU variants to calculate. Variants: \"cased\" "
+          "or \"uncased\"."))
+
+  FLAGS = flags.FLAGS
```

Comment:
Can you encapsulate everything the same way you did in transformer_main?

Reply:
What is the reason for encapsulating everything?

Reply:
In part it is conceptually easier if functions are pure, and in part because we may well want to use this in a way other than calling the file directly.

Reply:
Oooh I see. Thanks for the explanation.